### ISSUE TITLE
Clarify when parallel tasks are actually executed in parallel

### ISSUE THREAD

**Issue by pcmoen** [linkedin/parseq#63]
The documentation should be clarified on what's actually executed in parallel or how to ensure that tasks are executed in parallel.

The documentation have the following statement

> We can use `Task.par()` method to compose tasks to run in parallel

However when running the examples this doesn't seem true for the tasks composed of callables. Using `Task.par(callableTask1, callableTask2)` does not execute these tasks in parallel. This can be easily seen by viewing the Graphviz Timeline of the MergeSort example. None of the parallel operations are executed in parallel.

**Comment by jodzga** [comment]
I agree that documentation does not clarify this and it is probably most misunderstood aspect of parseq. I will provide better description on the wiki.

Thanks for bringing this up!

**Comment by jodzga** [comment]
Jvadoc should at least mention ParSeq execution model.
